src/crimson/os/seastore/onode.h

```diff
@@ -26,17 +26,20 @@ struct onode_layout_t {
   // TODO: implement flexible-sized onode value to store inline ss_attr
   // effectively.
   static constexpr int MAX_SS_LENGTH = 1;
+  static constexpr int MAX_OMAPHEADER_LENGTH = 256;
```

Comment:
We don't want to put this into the onode.  Most objects, even omap, don't have headers at all.  Sticking it into the omap keyspace is a problem since we pass user keys directly.  However, the osd prefixes user xattrs, so we can use unprefixed keys in the xattr key space freely.  Store omap header as an xattr with key "omap_header".

Reply:
Sam, xattr still saved in onode, oi, ss, xattr_root, all of them saved in onode layout. so what do you mean osd prefixes user xattrs?

Reply:
No, the only two xattr values we save inside of the onode are '_' (object_info) and 'snapset' (snapset) -- and even then only if they are smaller than MAX_OI_LENGTH and MAX_SS_LENGTH respectively.  All other xattr and omap values go into the xattr and omap trees.  Those two trees have root information in the onode, but the keys and values are in other non-onode btree leaf blocks.

Reply:
oh, got it, save it to xattr tree.

Reply:
done.